core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1691,15 +1691,12 @@ private[spark] object Utils extends Logging {
   /**
    * Default maximum number of retries when binding to a port before giving up.
    */
-  val portMaxRetries: Int = {
+  lazy val portMaxRetries: Int = {
     if (sys.props.contains("spark.testing")) {
       // Set a higher number of retries for tests...
       sys.props.get("spark.port.maxRetries").map(_.toInt).getOrElse(100)
     } else {
-      Option(SparkEnv.get)
-        .flatMap(_.conf.getOption("spark.port.maxRetries"))
-        .map(_.toInt)
-        .getOrElse(16)
+      sys.props.get("spark.port.maxRetries").map(_.toInt).getOrElse(16)
```

Comment:
I'm worried that changing this to read from System properties rather than the SparkEnv will break standalone mode here.  Have you confirmed that `spark.port.maxRetries` in standalone mode still works with this change?

In general we should prefer using SparkEnv over System properties because they have all kind of gotchas.  There's a reason we created SparkEnv in the first place.